app/jest/idp-approvals.test.tsx

```diff
@@ -0,0 +1,292 @@
+      Promise.resolve([[{ privacy_zone_uri: MOCK_PRIVACY_ZONE_URI, privacy_zone_name: MOCK_PRIVACY_ZONE_NAME }]]),
+    ),
+  };
+});
+
+const actionButtonsValidations = () => {
+  const deleteIcon = screen.queryByText(/Delete from Keycloak/i);
+  expect(deleteIcon).not.toBeInTheDocument();
+  const eventsIcon = screen.queryByText(/Events/i);
+  expect(eventsIcon).not.toBeInTheDocument();
+  const editIcon = screen.queryByText(/Edit/i);
+  expect(editIcon).not.toBeInTheDocument();
+  const restoreIcon = screen.queryByText(/Restore at Keycloak/i);
+  expect(restoreIcon).not.toBeInTheDocument();
+};
+
+describe('IDP Approvals', () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it('BCeID Approver', async () => {
+    jest
+      .spyOn(requestModule, 'getRequestAll')
+      .mockImplementationOnce(() => Promise.resolve([{ count: 1, rows: [sampleRequests()[0]] }, null]));
+    jest.spyOn(requestModule, 'updateRequest').mockImplementation(() => Promise.resolve([{}, null]));
+    jest
+      .spyOn(eventModule, 'getEvents')
+      .mockImplementation(() => Promise.resolve([{ count: 1, rows: sampleEvents() as any }, null]));
+    render(<AdminDashboard session={sampleSession} onLoginClick={jest.fn} onLogoutClick={jest.fn} />);
+
+    await waitFor(() => {
+      expect(screen.getByText('BCeID Approver')).toBeInTheDocument();
+    });
+
+    actionButtonsValidations();
+
+    fireEvent.click(screen.getByText('BCeID Approver'));
+    fireEvent.click(screen.getByText('BCeID Prod'));
+    const approveProdButton = screen.getByRole('button', { name: 'Approve Prod' });
+    fireEvent.click(approveProdButton);
+    expect(screen.getByText('Bceid Approve'));
+
+    jest
+      .spyOn(requestModule, 'getRequestAll')
+      .mockImplementationOnce(() =>
+        Promise.resolve([{ count: 1, rows: [{ ...sampleRequests()[0], bceidApproved: true }] }, null]),
+      );
+
+    //test on confirm button
+    fireEvent.click(screen.getByRole('button', { name: 'Confirm' }));
+    expect(updateRequest).toHaveBeenCalled();
+    await waitFor(() => {
+      expect(screen.queryByText('Bceid Approve')).not.toBeInTheDocument();
+    });
+
+    const approvedString = `Approved by ${sampleEvents()[0].idirUserDisplayName} on ${new Date(
+      sampleEvents()[0].createdAt,
+    ).toLocaleString()}`;
+
+    expect(screen.getByTestId('idp-approved-note')).toHaveTextContent(approvedString);
+  });
+
+  it('GitHub Approver', async () => {
+    jest
+      .spyOn(requestModule, 'getRequestAll')
+      .mockImplementationOnce(() => Promise.resolve([{ count: 1, rows: [sampleRequests()[1]] }, null]));
+    jest.spyOn(requestModule, 'updateRequest').mockImplementation(() => Promise.resolve([{}, null]));
+    jest
+      .spyOn(eventModule, 'getEvents')
+      .mockImplementation(() => Promise.resolve([{ count: 1, rows: sampleEvents() as any }, null]));
+    render(<AdminDashboard session={sampleSession} onLoginClick={jest.fn} onLogoutClick={jest.fn} />);
```

Comment:
Is the sample session changing to have the github role for this test, or does it still have the bceid-approver role only? I didn't see where it updates

Reply:
The sample session was just added to ensure users with approver roles are permitted to see admin dashboard as per new validation. The getRequestAll is mocked to return IDP specific requests but when deployed, the function uses approver roles to return IDP specific requests.

Reply:
Ah I see, sounds good!